app/src/main/java/com/celzero/bravedns/ui/SettingsFragment.kt

```diff
@@ -437,16 +460,17 @@ class SettingsFragment : Fragment(R.layout.activity_settings_screen) {
             }
         }
 
-
         b.settingsActivityExcludeAppsImg.setOnClickListener {
             b.settingsActivityExcludeAppsImg.isEnabled = false
             showExcludeAppDialog(requireContext(), excludeAppAdapter!!, excludeAppViewModel)
+            // the count-down-timer ticks once after 500ms, at the completion of which
+            // the settingsActivityExcludeAppsImg [is re-enabled again or add whatever is appropriate.
             object : CountDownTimer(500, 500) {
                 override fun onTick(millisUntilFinished: Long) {
                 }
 
                 override fun onFinish() {
-                    if(isAdded) {
+                    if (isAdded) {
```

Comment:
redundant `if` block. instead do:

```
  b.settingsActivityExcludedAppsImg.isEnabled = isAdded
```

And now... what ever `isAdded` means, it quite simply doesn't map onto `...isEnabled`. And so, rename the `var` as appropriate so that it reveals context.

Reply:
isAdded is part of Fragment. This is to check if the fragment is attached to the Activity. 
Also in onFinish(), the isEnabled should be true.

isAdded check is added because the UI element should not updated when the fragment is not added to the activity.

Reply:
In any case, "isAdded" is not the best name for this? Pls consider renaming it as appropriate. Same goes for whatever "isEnabled" needs to be renamed to, to make sure the name is appropriate in the context.

Reply:
isAdded is the object available in Fragment class not something we created.